test/core/test_stack_placement.c

```diff
@@ -22,7 +22,9 @@ int main(int argc, char* argv[]) {
   int* bss_address = &static_bss[BSS_BYTES-1];
   int* stack_address = &stack_var;
   printf("data: %p bss: %p stack: %p\n", data_address, bss_address, stack_address);
-  assert(stack_address > bss_address);
+  // Stack can either come after BSS or before data (In debug builds we link
+  // with --stack-first).
+  assert(stack_address > bss_address || stack_address < data_address);
```

Comment:
If there are such use cases we can do a followup that limits the impact of STACK_OVERFLOW_CHECK maybe?

Reply:
Sorry, I'm not sure what you mean, so maybe I wasn't clear. I was just saying that rather than checking `X || Y` we could check `X` if we expect the stack to be last and `Y` if we expect the stack to be first. And we could do `-DSTACK_FIRST` so the test knows what to check. So the check would be precise.

(If that doesn't clarify things, what do you mean by "use cases"?)

Reply:
Sorry .. replied to wrong comment.